### PR TITLE
travis: Test against Go 1.11, drop 1.9

### PR DESCRIPTION
**Pull request by abhinav**
Now that Go 1.11 is out, we should start testing against it. Plus to
match the [Go Release Policy][1], we are now free to drop support for
Go 1.9, which lets us rely on the built-in coverage merging functionality
instead of our hacky cover.sh script.

  [1]: https://golang.org/doc/devel/release.html#policy

### CODE REVIEW

**Comment by abhinav** [comment]
It looks like there's an issue in `go tool cover` in Go 1.11.

```
go tool cover -html=cover.out -o cover.html
cover: inconsistent NumStmt: changed from 2 to 3
```

This is reproducible locally. I'll try to figure out what's going on tomorrow.

**Comment by akshayjshah** [comment]
Seems reasonable, and I'm really excited to drop all our janky coverage-merging scripts. I'll be bummed out if there's really a regression in `go tool cover`.

**Comment by abhinav** [comment]
I've narrowed the issue down to the `//line` comments in the lex.go generated
by Ragel. Filed an issue (golang/go#27350) in case this is a regression.

Meanwhile, I'm going to filter out the `//line` clauses from the generated
files.

**Comment by abhinav** [comment]
I think I've driven the coverage as much as I can in this change. We can make
more follow up changes to improve coverage.

This is good to go @akshayjshah @AlexAPB @amckinney. Commits are individually
reviewable or all-in-one go based on your preference.

**Comment by abhinav** [comment]
To clarify: Previously, our coverage merging script was ignoring a number of
things when counting code coverage so we managed to get to ~85% with the
recent changes. With this change, `-coverpkg ./...` counts all Go code in
ThriftRW for coverage, including things that were previously ignored. We can
make follow-up changes to improve testing/coverage but I would rather not
block updating to the new compiler on that.

File: Makefile [hunk 1]
```diff
@@ -104,9 +104,24 @@ verifyversion: build
...
+
+# literal space
+space :=
+space +=
+
 .PHONY: cover
 cover:
-	go test -v -covermode=atomic -coverprofile cover.out -coverpkg=./... ./...
+	go test -v -covermode=atomic -coverprofile cover.full.out -coverpkg=./... ./...
+	grep -v "$(subst $(space),\|,$(COVER_IGNORE_FILES))" cover.full.out > cover.out
```
Comment (AlexAPB) [inline Makefile]: why do we need to replace spaces with `|`? nicer output?

Comment (abhinav) [reply Makefile]: We're building a regular expression so we need to replace the spaces with `|` clauses.

Comment (AlexAPB) [reply Makefile]: whoops, gotcha

**Pull request merged**